integrationTests/testProcessorNode.go

```diff
@@ -0,0 +1,715 @@
+		tpn.ArgsParser,
+		TestHasher,
+		TestMarshalizer,
+		tpn.AccntState,
+		tpn.BlockchainHook.(*hooks.VMAccountsDB),
+		TestAddressConverter,
+		tpn.ShardCoordinator,
+		tpn.ScrForwarder,
+	)
+
+	tpn.TxProcessor, _ = transaction.NewTxProcessor(
+		tpn.AccntState,
+		TestHasher,
+		TestAddressConverter,
+		TestMarshalizer,
+		tpn.ShardCoordinator,
+		tpn.ScProcessor,
+	)
+
+	fact, _ := shard.NewPreProcessorsContainerFactory(
+		tpn.ShardCoordinator,
+		tpn.Storage,
+		TestMarshalizer,
+		TestHasher,
+		tpn.ShardDataPool,
+		TestAddressConverter,
+		tpn.AccntState,
+		tpn.RequestHandler,
+		tpn.TxProcessor,
+		tpn.ScProcessor,
+		tpn.ScProcessor.(process.SmartContractResultProcessor),
+	)
+	tpn.PreProcessorsContainer, _ = fact.Create()
+
+	tpn.TxCoordinator, _ = coordinator.NewTransactionCoordinator(
+		tpn.ShardCoordinator,
+		tpn.AccntState,
+		tpn.ShardDataPool,
+		tpn.RequestHandler,
+		tpn.PreProcessorsContainer,
+		tpn.InterimProcContainer,
+	)
+}
+
+func (tpn *TestProcessorNode) initBlockProcessor() {
+	var err error
+
+	tpn.ForkDetector = &mock.ForkDetectorMock{
+		AddHeaderCalled: func(header data.HeaderHandler, hash []byte, state process.BlockHeaderState) error {
+			return nil
+		},
+		GetHighestFinalBlockNonceCalled: func() uint64 {
+			return 0
+		},
+		ProbableHighestNonceCalled: func() uint64 {
+			return 0
+		},
+	}
+
+	tpn.BlockTracker = &mock.BlocksTrackerMock{
+		AddBlockCalled: func(headerHandler data.HeaderHandler) {
+		},
+		RemoveNotarisedBlocksCalled: func(headerHandler data.HeaderHandler) error {
+			return nil
+		},
+		UnnotarisedBlocksCalled: func() []data.HeaderHandler {
+			return make([]data.HeaderHandler, 0)
+		},
+	}
+
+	if tpn.ShardCoordinator.SelfId() == sharding.MetachainShardId {
+		tpn.BlockProcessor, err = block.NewMetaProcessor(
+			&mock.ServiceContainerMock{},
+			tpn.AccntState,
+			tpn.MetaDataPool,
+			tpn.ForkDetector,
+			tpn.ShardCoordinator,
+			TestHasher,
+			TestMarshalizer,
+			tpn.Storage,
+			tpn.GenesisBlocks,
+			tpn.RequestHandler,
+			TestUint64Converter,
+		)
+	} else {
+		tpn.BlockProcessor, err = block.NewShardProcessor(
+			nil,
+			tpn.ShardDataPool,
+			tpn.Storage,
+			TestHasher,
+			TestMarshalizer,
+			tpn.AccntState,
+			tpn.ShardCoordinator,
+			tpn.ForkDetector,
+			tpn.BlockTracker,
+			tpn.GenesisBlocks,
+			true,
+			tpn.RequestHandler,
+			tpn.TxCoordinator,
+			TestUint64Converter,
+		)
+	}
+
+	if err != nil {
+		fmt.Printf("Error creating blockprocessor: %s\n", err.Error())
+	}
+}
+
+func (tpn *TestProcessorNode) setGenesisBlock() {
+	genesisBlock := tpn.GenesisBlocks[tpn.ShardCoordinator.SelfId()]
+	_ = tpn.BlockChain.SetGenesisHeader(genesisBlock)
+	hash, _ := core.CalculateHash(TestMarshalizer, TestHasher, genesisBlock)
+	tpn.BlockChain.SetGenesisHeaderHash(hash)
+}
+
+func (tpn *TestProcessorNode) initNode() {
+	var err error
+
+	if tpn.ShardCoordinator.SelfId() == sharding.MetachainShardId {
+		tpn.Node, err = node.NewNode(
+			node.WithMessenger(tpn.Messenger),
```

Comment:
use the common stuff with node.NewNode, than node.ApplyOptions for those which are different

Reply:
Good one. Will do.

Reply:
done